clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -75,6 +74,7 @@
     private final int autoCommitIntervalMs;
     private final ConsumerInterceptors<?, ?> interceptors;
     private final boolean excludeInternalTopics;
+    private final AtomicInteger pendingAsyncCommits;
```

Comment:
@hachikuji If coordinator disconnects and in-flight requests are aborted, `pendingAsyncCommits` will be decremented and the loop would no longer wait to find the new coordinator. If there are unsent commit requests, `pendingAsyncCommits` > 0 and the loop waits until a coordinator is known. I was trying to remove the inconsistency between auto-commit and manual async commit that Apurva pointed out. Isn't it worth attempting to send pending commits when possible?